bokeh/models/annotations.py

```diff
@@ -107,6 +108,89 @@ class Legend(Annotation):
         Dict(String, List(Instance(GlyphRenderer))), lambda d: list(d.items())
     )
 
+class Arrow(Annotation):
+    """ Render an arrow as an annotation.
+
+    """
+
+    tail_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow tails.
+    """)
+
+    tail_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow tails.
+    """)
+
+    head_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow heads.
+    """)
+
+    head_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow heads.
+    """)
+
+    head_size = Float(default=25, help="""
+    The size, in pixels, of the arrow head.
+    """)
+
+    tail_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the head_x and head_y attributes. Interpreted as "data
+    space" units by default.
+    """)
+
+    head_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the head_x and head_y attributes. Interpreted as "data
+    space" units by default.
+    """)
+
+    tail_style = Enum(ArrowStyle, default=None, help="""
+    The style of the head's arrow tail.
+    """)
+
+    head_style = Enum(ArrowStyle, default='open', help="""
+    The style of the head's arrow head.
+    """)
+
+    body_props = Include(LineProps, use_prefix=True, help="""
+    The %s values for the arrow body.
+    """)
+
+    tail_border_props = Include(LineProps, use_prefix=True, help="""
+    The %s values for the arrow tail outline.
+    """)
+
+    tail_body_props = Include(FillProps, use_prefix=True, help="""
+    The %s values for the arrow tail. If `fill_color` is set to None, the arrow
+    tail will be an open-backed wedge.
+    """)
+
+    tail_body_fill_color = Override(default="black")
+
+    head_border_props = Include(LineProps, use_prefix=True, help="""
+    The %s values for the arrow head outline.
+    """)
+
+    head_body_props = Include(FillProps, use_prefix=True, help="""
+    The %s values for the arrow head. If `fill_color` is set to None, the arrow
+    head will be an open-backed wedge.
+    """)
+
+    head_body_fill_color = Override(default="black")
```

Comment:
So first, I expect this will be a widely used interface, so we do have to "get it right" I see three options:
* add all the properties by hand (don't use `Include`)
* extend `Include` to accept a specified prefix in addition to inferring it
* split out a smaller `ArrowHead` class that `Arrow` can contain (then `Include` can be used on `ArrowHead` exactly as in glyph)

Under the last scenario, you'd have e.g., `arrow.head.line_color` and `arrow.tail.style`

I think I might be slightly partial to the last, actually, but I would accept any of the above.